### PR TITLE
show a bit more detail when finished precompiling

### LINKED ISSUE #55653: Names of startup packages that are being precompiled are not being printed anymore
Currently, on launching julia nightly, the output in the REPL is
```julia
$ julia +nightly
  6 dependencies successfully precompiled in 32 seconds. 16 already precompiled.
  8 dependencies successfully precompiled in 54 seconds. 12 already precompiled.
  4 dependencies successfully precompiled in 18 seconds. 10 already precompiled.
```
Previously, this used to print the names of the packages as well.
```julia
julia> versioninfo()
Julia Version 1.12.0-DEV.1127
Commit ca72e28f26b (2024-08-31 11:47 UTC)
Build Info:
  Official https://julialang.org release
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 8 × Intel(R) Core(TM) i5-10310U CPU @ 1.70GHz
  WORD_SIZE: 64
  LLVM: libLLVM-18.1.7 (ORCJIT, skylake)
Threads: 1 default, 0 interactive, 1 GC (on 8 virtual cores)
Environment:
  JULIA_EDITOR = subl
```

### PR DESCRIPTION
**Pull request by IanButterworth**
Fixes https://github.com/JuliaLang/julia/issues/55653

Screenshot 2024-08-31 at 9 25 43 PM

---

Screenshot 2024-08-31 at 9 26 00 PM

### CODE REVIEW

**Comment by KristofferC** [comment]
To me, this looks like it is still precompiling since it has a header that says so.

**Comment by IanButterworth** [comment]
The issue is it can't just say precompiled because it could have failed.

File: base/precompilation.jl [hunk 1]
```diff
@@ -919,10 +920,16 @@ function precompilepkgs(pkgs::Vector{String}=String[];
     seconds_elapsed = round(Int, (time_ns() - time_start) / 1e9)
     ndeps = count(values(was_recompiled))
     if ndeps > 0 || !isempty(failed_deps) || (quick_exit && !isempty(std_outputs))
-        str = sprint() do iostr
+        str = sprint(context=io) do iostr
             if !quick_exit
+                if fancyprint # replace the progress bar
+                    packages = isempty(requested_pkgs) ? "packages." : "$(join(requested_pkgs, ", "))."
+                    printpkgstyle(iostr, :Precompiling, packages, newline=false)
```
Comment (IanButterworth) [inline base/precompilation.jl:927]: @KristofferC what about
```suggestion
                    printpkgstyle(iostr, :Precompiling, "finished. " * packages, newline=false)
```

**Comment by IanButterworth** [comment]
@KristofferC what about this?
Screenshot 2024-09-03 at 10 56 20 AM

**Pull request merged**